sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -90,5 +92,39 @@ public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName
             States = states;
             Headers = headers;
         }
+
+        /// <summary>
+        /// Method to deserialize state to customized type.
+        /// </summary>
+        /// <typeparam name="T">Customized type.</typeparam>
+        /// <param name="stateKey">State key.</param>
+        /// <param name="value">Connection state mapped to the key.</param>
+        /// <returns>Returns true if is able to find the key and false means key doesn't exist or value type not able to convert.</returns>
+        public bool TryGetState<T>(string stateKey, out T value)
+        {
+            value = default(T);
+            if (States.TryGetValue(stateKey, out var stateValue))
+            {
+                if (stateValue is T val)
+                {
+                    value = val;
+                    return true;
+                }
+                // Should ensure in server SDK.
+                if (stateValue is JsonElement element)
+                {
+                    try
+                    {
+                        value = JsonSerializer.Deserialize<T>(element.GetRawText());
```

Comment:
Can you use the overload that starts with a `JsonElement` instead of going back to raw text?  https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-6.0#System_Text_Json_JsonSerializer_Deserialize__1_System_Text_Json_JsonElement_System_Text_Json_JsonSerializerOptions_

Reply:
This looks great, however, we're NETStandard2.0 to support functions major versions and not able to use this.